### PR TITLE
MSSQL Fixes

### PR DESCRIPTION
**Pull request by lcamilo**

### CODE REVIEW

File: hmsUpdateChartTables.ps1 [hunk 1]
```diff
@@ -152,11 +154,14 @@ $Query = "
...
 		FROM hm_fwban_rh 
-		GROUP BY day, hour
+		GROUP BY $( DBCastDateTimeFieldAsDate 'timestamp'), $( DBCastDateTimeFieldAsHour 'timestamp')
 	) d 
 	GROUP BY hour 
 	ORDER BY hour ASC
 "
+
+Write-Host $Query
+
```
Comment (palinkas-jo-reggelt) [inline hmsUpdateChartTables.ps1:164]: This one runs unattended. No need for write-host. Nobody will ever see it.

Comment (lcamilo) [reply hmsUpdateChartTables.ps1:164]: Sorry.

Comment (palinkas-jo-reggelt) [reply hmsUpdateChartTables.ps1:164]: Oops - I just realized that was debugging...

**Comment by palinkas-jo-reggelt** [comment]
I have a question about this:

```($( DBFormatDate (DBCastDateTimeFieldAsDate 'timestamp') '%c') $(If ($DatabaseType -eq 'MYSQL'){Write "- 1"})) AS month,```

The reason I subtract 1 from the month is because JavaScript month format is 0-11, so its necessary for the charts. Were you aware of that? I left it alone because I thought maybe SQL server does something automatically, but after thinking about it, I figure you may not be aware of the reason its there.

**Comment by lcamilo** [comment]
> The reason I subtract 1 from the month is because JavaScript month format is 0-11, so its necessary for the charts. Were you aware of that? I left it alone because I thought maybe SQL server does something automatically, but after thinking about it, I figure you may not be aware of the reason its there.



> I have a question about this:
> 
> `($( DBFormatDate (DBCastDateTimeFieldAsDate 'timestamp') '%c') $(If ($DatabaseType -eq 'MYSQL'){Write "- 1"})) AS month,`
> 
> The reason I subtract 1 from the month is because JavaScript month format is 0-11, so its necessary for the charts. Were you aware of that? I left it alone because I thought maybe SQL server does something automatically, but after thinking about it, I figure you may not be aware of the reason its there.

Ow.. I know about javascript, but I thought this was some adjustment because of MySql. Sorry, as I don't use MySql, I thought it was a fix because of it.

That way, I'll check if we need to make an adjustment in MSSQL, because of data types (string varchar plus an int).

**Comment by palinkas-jo-reggelt** [comment]
> > The reason I subtract 1 from the month is because JavaScript month format is 0-11, so its necessary for the charts. Were you aware of that? I left it alone because I thought maybe SQL server does something automatically, but after thinking about it, I figure you may not be aware of the reason its there.
> 
> > I have a question about this:
> > `($( DBFormatDate (DBCastDateTimeFieldAsDate 'timestamp') '%c') $(If ($DatabaseType -eq 'MYSQL'){Write "- 1"})) AS month,`
> > The reason I subtract 1 from the month is because JavaScript month format is 0-11, so its necessary for the charts. Were you aware of that? I left it alone because I thought maybe SQL server does something automatically, but after thinking about it, I figure you may not be aware of the reason its there.
> 
> Ow.. I know about javascript, but I thought this was some adjustment because of MySql. Sorry, as I don't use MySql, I thought it was a fix because of it.
> 
> That way, I'll check if we need to make an adjustment in MSSQL, because of data types (string varchar plus an int).

LOL we both thought the same thing. I didn't say anything before because I though MSSQL handled it automatically. 

This -1 only applies to the charts. I'll fix it in the powershell chart updater.

**Pull request merged**